cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -555,7 +555,7 @@ predicate instructionForFullyConvertedCall(Instruction instr, CallInstruction ca
 }
 
 /** Holds if `node` represents the output node for `call`. */
-private predicate simpleOutNode(Node node, CallInstruction call) {
+predicate simpleOutNode(Node node, CallInstruction call) {
```

Comment:
Is this the correct way to go about this, or should the predicate actually be moved to `DataFlowUtil`. Note that I'm not really sure about the difference between the two.

Reply:
`DataFlowUtil` is supposed to be "things that you're actually allowed to use in queries" and `DataFlowPrivate` is supposed to be "private things that's users are not supposed to know about".

You can see the distinction between the two files in the way they're imported by the dataflow library here: https://github.com/github/codeql/blob/main/cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImplSpecific.qll#L7. As you can see, `DataFlowPrivate` is imported in a module `Private`, and `DataFlowUtil` is imported in a module `Public`. _Somewhere_ (I don't actually know where) that `Public` module is publicly imported so that it's available when you import the dataflow library in a query.

So I think this is actually the correct thing to do. Since everything in `DataFlowPrivate` is never made public at the query level, removing the `private` keyword here doesn't really change what's visible outside the dataflow library. It just means that other dataflow implementations files can access it.

Reply:
I should say, it took me a couple of years to realize what I wrote about about the distinction between `DataFlowUtil` and `DataFlowPrivate` so I thoroughly misplaced many classes and predicates until then 😂.